### PR TITLE
Removes endpoint to get L1 height from Obs node.

### PR DESCRIPTION
**Pull request by joeldudley**
### Why is this change needed?

We were offering an RPC endpoint on the Obscuro node for test purposes, to get the L1 chain height. This didn't make sense - the sims can just grab the L1 height from the L1. So I deleted the endpoint. In turn, this allowed me to simplify the host's DB logic for blocks.

### What changes were made as part of this PR:

- Provide a high level list of the changes made
- List key areas for the reviewer 

### :rotating_light: Definition of done :rotating_light:
- [ ] Unit tests added to cover new or changed functionality 
- [ ] Docs pages updated to cover new or changed functionality
- [ ] [Changelog.md](https://github.com/obscuronet/go-obscuro/blob/main/docs/testnet/changelog.md) updated

### CODE REVIEW

File: go/ethadapter/blockprovider_test.go [hunk 1]
```diff
@@ -171,6 +171,11 @@ func (e *ethClientMock) BlockByNumber(num *big.Int) (*types.Block, error) {
 	return block, nil
 }
 
+func (e *ethClientMock) BlockNumber() (uint64, error) {
```
Comment (joeldudley) [inline go/ethadapter/blockprovider_test.go:174]: This is a new method we added to L1 clients, so we have to implement it on the mock clients now.

File: go/host/db/blocks.go [hunk 2]
```diff
@@ -13,16 +12,7 @@ import (
 	"github.com/ethereum/go-ethereum/rlp"
 )
 
-// DB methods relating to batches.
-
-// GetHeadBlockHeader returns the block header of the current head block.
-func (db *DB) GetHeadBlockHeader() (*types.Header, error) {
```
Comment (joeldudley) [inline go/host/db/blocks.go:19]: Storing and retrieving the head block is no longer required - we only did it to power that test endpoint.

File: go/host/db/blocks_test.go [hunk 3]
```diff
@@ -41,68 +41,3 @@ func TestUnknownBlockHeaderReturnsNotFound(t *testing.T) {
 		t.Errorf("did not store block header but was able to retrieve it")
 	}
 }
-
-func TestHigherNumberBlockBecomesBlockHeader(t *testing.T) {
```
Comment (joeldudley) [inline go/host/db/blocks_test.go:45]: Tests of deleted method.

File: go/host/db/hostdb.go [hunk 4]
```diff
@@ -15,16 +15,15 @@ import (
 
 // Schema keys, in alphabetical order.
 var (
-	headBlock            = []byte("hb")
```
Comment (joeldudley) [inline go/host/db/hostdb.go:18]: I alphabetised these.

File: go/host/rpc/clientapi/client_api_obscuro.go [hunk 5]
```diff
@@ -4,8 +4,6 @@ import (
 	"github.com/obscuronet/go-obscuro/go/common/host"
 )
 
-// TODO - Some methods return nil for an unfound block/rollup, while others return an error. Harmonise.
```
Comment (joeldudley) [inline go/host/rpc/clientapi/client_api_obscuro.go:7]: This was no longer true.

**Review by tudor-malene — APPROVED** [review]
lgtm

**Pull request merged**